models/vrp.rb

```diff
@@ -391,6 +391,16 @@ def self.remove_unnecessary_relations(hash)
         uniq_relations += relations_set.uniq
       }
       hash[:relations] = uniq_relations
+
+      return unless hash[:configuration][:preprocessing].to_h[:partitions].to_a.none?{ |p| p[:entity] == :vehicle } &&
```

Comment:
if+any will avoid the double negation here
```suggestion
      return if hash[:configuration][:preprocessing].to_h[:partitions].to_a.any?{ |p| p[:entity] == :vehicle } &&
```